rest-api/dao/base_dao.py

```diff
@@ -154,28 +154,28 @@ def _from_json_value(self, prop, value):
   def query(self, query_def):
     if not self.order_by_ending:
       raise BadRequest("Can't query on type %s -- no order by ending speciifed" % self.model_type)
+
     with self.session() as session:
       query, field_names = self._make_query(session, query_def)
       items = query.all()
-    if items:
-      if len(items) > query_def.max_results:
-        # Items, pagination token, and more are available
-        return Results(items[0:query_def.max_results],
-                       self._make_pagination_token(items[query_def.max_results - 1].asdict(),
-                                                   field_names),
-                       True)
-      else:
-        if query_def.always_return_token:
-          # Items and pagination token, but no more available
-          return Results(items,
-                         self._make_pagination_token(items[len(items) - 1].asdict(), field_names),
-                         False)
-        else:
-          # Items but no pagination token, and no more available
-          return Results(items, None, False)
+
+      if not items:
```

Comment:
Two concerns about this:

1) If query_def.pagination_token is not None, we could be returning a 0 total when there clearly are some results. (We paginated through them.)

2) You're returning 0 here even when query_def.include_total is False... in other cases total is None in that situation.

Reply:
Good call on both counts - 2 I had expected b/c the length of an empty list is 0, but it _is_ inconsistent. Fixed.